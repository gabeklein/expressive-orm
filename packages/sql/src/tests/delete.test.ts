import { Query, Str, Type } from '../';

class Foo extends Type {
  value = Str();
  color = Str();
}

class Bar extends Type {
  value = Str();
  color = Str();
}

it("will generate query", () => {
  const query = Query(where => {
    const foo = where(Foo);

    where(foo.value).is("Hello World!");
    where(foo).delete();
  });

  expect(query).toMatchSnapshot();
})

it("will include FROM statement where JOIN exists", () => {
  const query = Query(where => {
    const foo = where(Foo);
    const bar = where(Bar, { color: foo.color });

    where(bar.value).is("Hello World!");
    where(foo).delete();
  });

  expect(query).toMatchSnapshot();
})

// this is only suprted in MySQL
// it.skip("will delete from multiple", () => {
//   const query = Query(where => {
//     const foo = where(Foo);
//     const bar = where(Bar, { color: foo.color });

//     where(bar.value).is("Hello World!");

//     where.delete(foo, bar);
//   });

//   expect(query).toMatchSnapshot();
// })