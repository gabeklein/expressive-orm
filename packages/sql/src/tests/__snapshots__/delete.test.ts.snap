// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will generate query 1`] = `
	delete from
	  \`foo\`
	where
	  \`foo\`.\`value\` = 'Hello World!'
`;

exports[`will include FROM statement where JOIN exists 1`] = `
	delete \`foo\`
	from
	  \`foo\`
	  inner join \`bar\` on \`bar\`.\`color\` = \`foo\`.\`color\`
	where
	  \`bar\`.\`value\` = 'Hello World!'
`;
